### ISSUE TITLE
CALM meta-schema use of 'required' is incorrect according to JSON schema standard

### ISSUE THREAD

**Issue by willosborne** [finos-labs/architecture-as-code#75]
## Bug Report

The CALM meta-schema specifies 'nodes' and 'relationships' as top-level required fields here: 

https://github.com/finos-labs/architecture-as-code/blob/main/calm/draft/2024-03/meta/core.json#L18

There are two issues with this:
- It is specified inside the list of properties, instead of outside the list. 
- Specifying it outside the list does not have the intended effect.

In JSON meta-schema it does not appear to be possible to mandate that all schemas have a top-level property set.
If the required block was to be moved out of the properties list - in order to make the schema syntactically correct per the first issue - what this configuration does currently is **specify that all objects defined must have nodes and relationships set on them**

In other words, this applies to objects one level down in the schema, rather than the top-level objects as intended.
This means that code generation and validation gets confused, and reports that 'nodes' is missing on inner objects. It also generates extra nodes/relationships properties that should not be present.

### Intended fix
Remove the required nodes/relationships section.
This will instead be enforced by Spectral validation rules.

**Comment by rocketstack-matt** [comment]
This issue is resolved, addition of Spectral rules for pattern validation that the nodes and relationships are present will be tracked in #95
